SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -45,12 +45,23 @@ TMap<FString, uint32> LevelPathToComponentId;
 TSet<uint32> LevelComponentIds;
 
 // Prevent name collisions.
-TMap<UClass*, FString> ClassToSchemaName;
-TMap<FString, UClass*> UsedSchemaNames;
+TMap<FString, FString> ClassToSchemaName;
+TMap<FString, FString> UsedSchemaNames;
+TMap<FString, TSet<FString>> NameCollisions;
 
 namespace
 {
 
+void AddNameCollision(FString SchemaName, FString ClassPath)
```

Comment:
This is for 'potential' collisions right? Maybe "AddSchemaNameUsage" or "AddUsedSchemaName"